### PR TITLE
List of users

### PR DESCRIPTION
**Pull request by juada96**
## Summary

* service to access the list of users registered 
*in order to have access to the list, the user must be logged In 
*there is a pagination which separes every 5 results 

## Known Issues
*I'm still not sure how to test with active session, however, the service works perfectly

## Trello Card
https://trello.com/c/HB1eCs5G/4-lista-de-usuarios

### CODE REVIEW

File: app/controllers/userList.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+const _user = require('./../models').user;
+
+exports.userList = (req, res) => {
+  const usermail = req.body.email;
+  if (_user.findAll({ attributes: ['sesion'] }, { where: { email: usermail } })) {
```
Comment (gonziesc) [inline app/controllers/userList.js]: why this if?

Comment (gonziesc) [reply app/controllers/userList.js]: also, return

File: app/controllers/userList.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    _user
```
Comment (gonziesc) [inline app/controllers/userList.js]: return

File: app/controllers/userList.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+      .findAll({ attributes: ['name', 'lastName', 'email', 'role'] }, { limit: 5 })
+      .then(userlist => {
+        res.json(userlist).end();
```
Comment (gonziesc) [inline app/controllers/userList.js]: what about the pagination?

File: app/controllers/userList.js (hunk 1 above)
Comment (juada96) [reply app/controllers/userList.js]: according to the trello card's criteria : "Se debe estar autenticado para consumir dicho recurso."
  
so the field 'sesion' is a boolean which is set on true right after performing a sign in, in theory, if this field returns true, it means the user who is trying to access the service is active.

File: app/controllers/userList.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
+const _user = require('./../models').user;
+
+exports.userList = async (req, res) => {
+  const usermail = req.body.email;
+  if (await _user.findAll({ attributes: ['sesion'] }, { where: { email: usermail } }).data) {
```
Comment (gonziesc) [inline app/controllers/userList.js]: no need of this if

File: app/controllers/userList.js [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    const limit = req.body.limit;
+    const pages = req.body.page;
+    return _user
+      .findAndCountAll(
+        { attributes: ['name', 'lastName', 'email', 'role'] },
+        { limit, offset: limit * (pages - 1) }
+      )
+      .then(userlist => {
+        const numberOfUsers = userlist.count;
```
Comment (gonziesc) [inline app/controllers/userList.js]: you have to count all, not just the limit you defined

File: app/controllers/userController.js [hunk 6]
```diff
@@ -0,0 +1,29 @@
+const _user = require('./../models').user,
+  jwt = require('jsonwebtoken'),
+  secretk = require('../../config/index').config;
+
+const exist = usermail => {
+  const user = _user.findAll({ where: { email: usermail } });
+  return user == null;
```
Comment (gonziesc) [inline app/controllers/userController.js]: please dont compare to null

File: app/controllers/userController.js [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+};
+
+const activeSesion = async req => {
+  const usermail = req.body.email;
+  const active = await _user.findAll({ attributes: ['sesion'] }, { where: { email: usermail } });
+  return active;
```
Comment (gonziesc) [inline app/controllers/userController.js]: this is not necessary by now, just check that the recieved token is ok

File: app/controllers/userController.js [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+};
+
+const isAdmin = async user => {
+  const usermail = user.email;
+  const role = await _user.findAll({ attributes: ['role'] }, { where: { email: usermail } });
```
Comment (gonziesc) [inline app/controllers/userController.js]: you are repeating a lot this lines

File: app/controllers/userController.js [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+  return role === 'admin';
+};
+
+const authenticated = req => {
+  if (activeSesion(req)) {
+    const token = req.cookie;
+    return jwt.verify(token, secretk.session.secret).then(decoded => {
```
Comment (gonziesc) [inline app/controllers/userController.js]: better

File: app/controllers/userController.js [hunk 10]
```diff
@@ -0,0 +1,29 @@
+const _user = require('./../models').user,
```
Comment (gonziesc) [inline app/controllers/userController.js]: you are not exporting nothing from this file

File: app/controllers/userController.js [hunk 11]
```diff
@@ -0,0 +1,23 @@
...
+  jwt = require('jsonwebtoken'),
+  secretk = require('../../config/index').config;
+
+const exist = usermail => {
+  return _user.findAll({ where: { email: { $contains: [usermail] } } }).then(user => {
+    !!user;
```
Comment (gonziesc) [inline app/controllers/userController.js]: return

File: app/controllers/userController.js [hunk 12]
```diff
@@ -0,0 +1,23 @@
...
+  });
+};
+
+const isAdmin = user => {
+  const usermail = user.email;
+  return _user.findAll({ params: ['role'] }, { where: { email: usermail } }).then(role => {
+    role === 'admin';
```
Comment (gonziesc) [inline app/controllers/userController.js]: return..

File: app/controllers/userController.js [hunk 13]
```diff
@@ -0,0 +1,23 @@
...
+  });
+};
+
+const authenticated = req => {
+  const token = req.cookie;
+  return jwt.verify(token, secretk.session.secret).then(decoded => {
+    exist(decoded);
```
Comment (gonziesc) [inline app/controllers/userController.js]: return

File: app/routes.js [hunk 14]
```diff
@@ -1,7 +1,9 @@
-const signIn = require('./controllers/singIn');
+const signIn = require('./controllers/singIn'),
+  listOfUsers = require('./controllers/userList');
 
 exports.init = app => {
   // app.get('/endpoint/get/path', [], controller.methodGET);
   // app.put('/endpoint/put/path', [], controller.methodPUT);
   app.post('/user', signIn.signUp);
+  app.get('/listOfUsers', listOfUsers.userList);
```
Comment (gonziesc) [inline app/routes.js]: /users

**Pull request merged**